### PR TITLE
New internal user request page

### LINKED ISSUE #1122: New internal user request page

### PR DESCRIPTION
**Pull request by nyet-ty**
Resolve #1122 

## PR includes

-   [x] Feature

### CODE REVIEW

File: src/components/InternalUserCreationRequestPage/InternalUserCreationRequestPage.tsx [hunk 1]
```diff
@@ -0,0 +1,469 @@
...
+        if (isLoginUnique.data === false) {
+            setError('login', { message: tr('User with login already exist') });
+        } else clearErrors('login');
+    }, [isLoginUnique.data, setError, clearErrors]);
+
+    const debouncedSearchHandler = debounce(300, setIsLoginUniqueQuery);
+
+    const onNameChange = () => {
+        const login = loginAuto({
+            firstName: watch('firstName'),
```
Comment (tsumo) [inline src/components/InternalUserCreationRequestPage/InternalUserCreationRequestPage.tsx:104]: getValue is better for one-time use

File: src/components/PageHeaderActionButton/PageHeaderActionButton.tsx [hunk 2]
```diff
@@ -2,28 +2,29 @@ import { useMemo } from 'react';
...
-    const createUserModalVisibility = useBoolean(false);
+    const router = useRouter();
+
     const createGroupModalVisibility = useBoolean(false);
 
     const items: { title: string; action: VoidFunction }[] = useMemo(() => {
         const result = [];
         if (sessionUser.role?.createUser) {
-            result.push({ title: tr('User'), action: createUserModalVisibility.setTrue });
+            result.push({ title: tr('User'), action: () => router.newInternalUserRequest() });
```
Comment (tsumo) [inline src/components/PageHeaderActionButton/PageHeaderActionButton.tsx:23]: Don't need arrow func wrapper here

File: src/modules/userCreationRequestSchemas.ts [hunk 3]
```diff
@@ -26,24 +33,45 @@ export type CreateUserCreationRequestBase = z.infer<typeof createUserCreationReq
...
-    location: z.string().min(1, { message: tr('Minimum {min} symbols', { min: 1 }) }),
+    workEmail: z.string().email(tr('Not a valid email')).optional().or(z.literal('')),
+    location: z.string().min(1, { message: tr('Required field') }),
     creationCause: z.string(),
     unitId: z.string().optional(),
-    date: z.date(),
+    date: z.date({ invalid_type_error: tr('Required field'), required_error: tr('Required field') }),
+    percentage: z
+        .number({ required_error: tr('Please enter percentage from 0.01 to 1') })
+        .multipleOf(0.01)
```
Comment (tsumo) [inline src/modules/userCreationRequestSchemas.ts:60]: cool!

**Review by tsumo — CHANGES_REQUESTED** [review]

**Review by tsumo — APPROVED** [review]

**Review by asabotovich — CHANGES_REQUESTED** [review]

File: src/components/RoleSelect/RoleSelect.tsx [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    return (
+        <Select
+            arrow
+            value={value}
+            items={rolesQuery.data}
+            onChange={(roles) => onChange(roles[0])}
+            selectable
+            mode="single"
+            renderItem={(props) => (
+                <MenuItem key={props.item.id}>
```
[the remark this answers is not in the document]
Comment (nyet-ty) [reply src/components/RoleSelect/RoleSelect.tsx]: corrected in all Selects

**Pull request merged**